src/tinode.js

```diff
@@ -3543,7 +3543,21 @@ Topic.prototype = {
     pub._failed = false;
     return this._tinode.publishMessage(pub).then((ctrl) => {
       pub._sending = false;
-      pub.seq = ctrl.params.seq;
+      let idx = this._messages.find({
+        seq: pub.seq
+      });
+      let newSeq = ctrl.params.seq;
+      let numMessages = this._messages.length();
+      pub.seq = newSeq;
+      if (0 <= idx && idx < numMessages) {
+        // this._messages are sorted by `seq`.
+        // If changing pub.seq to newSeq breaks the invariant, fix it.
+        if ((idx > 0 && this._messages.getAt(idx - 1).seq >= newSeq) ||
+          (idx + 1 < numMessages && newSeq < this._messages.getAt(idx + 1).seq <= newSeq)) {
+          this._messages.delAt(idx);
+          this._messages.put(pub);
+        }
+      }
```

Comment:
Loolks like you are editing some older version of tinode-js (master?).  This has been fixed a while ago:
https://github.com/tinode/tinode-js/commit/7b84cce6d3b48e08e1e157cab55551216773242c

Reply:
Moved this code to swapMessage().